src/encoder.rs

```diff
@@ -1110,27 +1110,32 @@ pub fn encode_tx_block<T: Pixel>(
   cw: &mut ContextWriter,
   w: &mut dyn Writer,
   p: usize,
+  // Offset in the luma plane of the partition enclosing this block.
   tile_partition_bo: TileBlockOffset,
-  bx: usize,
-  by: usize, // tx block position within a partition, unit: tx block number
-  tile_bo: TileBlockOffset,
+  // tx block position within a partition, unit: tx block number
+  bx: usize, by: usize,
+  // Offset in the luma plane where this tx block is colocated. Note that for
+  // a chroma block, this offset might be outside of the current partition.
+  // For example in 4:2:0, four 4x4 luma partitions share one 4x4 chroma block,
+  // this block is part of the last 4x4 partition, but its `tx_bo` offset
+  // matches the offset of the first 4x4 partition.
+  tx_bo: TileBlockOffset,
   mode: PredictionMode,
   tx_size: TxSize,
   tx_type: TxType,
   bsize: BlockSize,
   po: PlaneOffset,
   skip: bool,
+  qidx: u8,
   ac: &[i16],
   alpha: i16,
   rdo_type: RDOType,
   need_recon_pixel: bool,
 ) -> (bool, ScaledDistortion) {
-  let qidx = get_qidx(fi, ts, cw, tile_bo);
```

Comment:
So, for chroma block with sub8x8 partition, instead of tile_bo it should be the last sub8x8 luma block.